### ISSUE TITLE
IORING_OP_READ for eventfd

### ISSUE THREAD

**Issue by cor3ntin** [axboe/liburing#59]
I've been trying and failing to read from a fd opened with `eventfd` through `IORING_OP_READ`

```
int fd = eventfd(0, 0);
io_uring_prep_read(sqe, fd, &event, sizeof(eventfd_t), 0);
```
This fails consistently with `EINVAL`
Polling with `io_uring_prep_poll_add(sqe, fd, 0);` then reading with `read(fd, &event, sizeof(eventfd_t))` works.

am I missing something or is reading for eventfd directly through io_uring not supported ?

Thanks

**Comment by axboe** [comment]
Can you post your test case? Saves me from having to rewrite one.

**Comment by CarterLi** [comment]
Which version of kernel are you using? Have you tried `IORING_OP_READV`?

**Comment by cor3ntin** [comment]
Here is a test
https://github.com/axboe/liburing/pull/60

**Comment by axboe** [comment]
You're probably running it on a kernel that doesn't support IORING_OP_READ - the non-vectored read/write commands are only going to be available in the 5.6 kernel, not available in 5.5 or earlier. I ran your test program on my 5.6 branch (for-5.6/io_uring-vfs) and it works just fine, whereas it'll expectedly -EINVAL on 5.5 and earlier.

**Comment by cor3ntin** [comment]
I had a for-5.6/io_uring from december - I switched to for-5.6/io_uring-vfs and it works.
I guess I'll use that for now, thanks
